pkg/controller/miganalytic/miganalytics_controller.go

```diff
@@ -0,0 +1,589 @@
+	} else {
+		return reconcile.Result{RequeueAfter: time.Second * RequeueInterval}, nil
+	}
+
+	// Ready
+	analytic.Status.SetReady(
+		!analytic.Status.HasBlockerCondition(),
+		ReadyMessage)
+
+	// End staging conditions.
+	analytic.Status.EndStagingConditions()
+
+	// Apply changes.
+	analytic.MarkReconciled()
+	err = r.Update(context.TODO(), analytic)
+	if err != nil {
+		log.Trace(err)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Done
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileMigAnalytic) analyze(analytic *migapi.MigAnalytic) error {
+	plan := &migapi.MigPlan{}
+
+	err := r.getAnalyticPlan(analytic, plan)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	srcCluster, err := plan.GetSourceCluster(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	client, err := srcCluster.GetClient(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	discovery, err := discovery.NewDiscoveryClientForConfig(client.RestConfig())
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	resources, err := collectResources(discovery)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	dynamic, err := dynamic.NewForConfig(client.RestConfig())
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	analytic.Status.Analytics.Plan = plan.Name
+	log.Info("Beginning analysis of plan.", "migplan", plan.Name)
+
+	for i, namespace := range plan.Spec.Namespaces {
+		log.Info("Beginning analysis of namespace.", "namespace", namespace)
+		ns := migapi.MigAnalyticNamespace{
+			Namespace: namespace,
+		}
+
+		excludedResources := plan.Status.ExcludedResources
+		incompatible := plan.Status.Incompatible
+
+		if analytic.Spec.AnalyzeK8SResources {
+			err := r.analyzeK8SResources(dynamic, resources, &ns, excludedResources, incompatible)
+			if err != nil {
+				return liberr.Wrap(err)
+			}
+		}
+		if analytic.Spec.AnalyzeImageCount {
+			err := r.analyzeImages(client, &ns, analytic.Spec.ListImages, analytic.Spec.ListImagesLimit)
+			if err != nil {
+				return liberr.Wrap(err)
+			}
+		}
+		if analytic.Spec.AnalyzePVCapacity {
+			err := r.analyzePVCapacity(client, &ns)
+			if err != nil {
+				return liberr.Wrap(err)
+			}
+		}
+
+		analytic.Status.Analytics.Namespaces = append(analytic.Status.Analytics.Namespaces, ns)
+		analytic.Status.Analytics.K8SResourceTotal += ns.K8SResourceTotal
+		analytic.Status.Analytics.ExcludedK8SResourceTotal += ns.ExcludedK8SResourceTotal
+		analytic.Status.Analytics.IncompatibleK8SResourceTotal += ns.IncompatibleK8SResourceTotal
+		analytic.Status.Analytics.ImageCount += ns.ImageCount
+		analytic.Status.Analytics.ImageSizeTotal.Add(ns.ImageSizeTotal)
+		analytic.Status.Analytics.PVCapacity.Add(ns.PVCapacity)
+		analytic.Status.Analytics.PVCount += ns.PVCount
+		analytic.Status.Analytics.PercentComplete = (i + 1) * 100 / len(plan.Spec.Namespaces)
+
+		err = r.Update(context.TODO(), analytic)
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+		log.Info("Completed analysis of namespace.", "namespace", namespace)
```

Comment:
Recommend we remove these "Completed" messages as well. If we want to tell a user which namespaces were completed, I would advise including them as a `Condition` on the `Status.Conditions` array.

Reply:
This is in an effort to keep the mig-controller logs uncluttered, except for when critical errors occur.